main.go

```diff
@@ -135,3 +319,183 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
+			if err != nil {
+				logrus.Errorf("An error occurred while converting tags to semantic version.", err)
+			}
+			tagSemVers = append(tagSemVers, tagSemVer) 
+		}
+		return nil
+	})
+
+	var latestReleaseTagSemVer *semver.Version
+	if len(tagSemVers) == 0 {
+		latestReleaseTagSemVer, err = semver.StrictNewVersion(NO_PREVIOUS_VERSION)
+		if err != nil {
+			logrus.Errorf("An error occurred while converting tags to semantic version.", err)
+		}
+	} else {
+		sort.Sort(sort.Reverse(semver.Collection(tagSemVers)))
+		latestReleaseTagSemVer = tagSemVers[0]
+	}
+
+	return *latestReleaseTagSemVer
+}
+
+func detectBreakingChanges(changelogFilepath string) bool {
+	changelogFile, err := os.Open(changelogFilepath);
+	if err != nil {
+		logrus.Errorf("Error attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath, err)
+	}
+	defer changelogFile.Close()
+	tbdRegex, err := regexp.Compile(TBD_VERSION_HEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", TBD_VERSION_HEADER_REGEX, err)
+	}
+	breakingChangesRegex, err := regexp.Compile(BREAKING_CHANGES_SUBHEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", BREAKING_CHANGES_SUBHEADER_REGEX, err)
+	}
+	versionHeaderRegex, err := regexp.Compile(VERSION_HEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", VERSION_HEADER_REGEX, err)
+	}
+
+    scanner := bufio.NewScanner(changelogFile)
+	// Find TBD header
+    for scanner.Scan() {
+		if tbdRegex.Match(scanner.Bytes()) {
+			break
+		}
+    }
+	// Scan file until next version header detected, searching for Breaking Changes header along the way
+	foundBreakingChanges := false
+	for scanner.Scan() {
+		if breakingChangesRegex.Match(scanner.Bytes()){
+			foundBreakingChanges = true
+		}
+		
+		if versionHeaderRegex.Match(scanner.Bytes()){
+			break
+		}
+	}
+
+    if err := scanner.Err(); err != nil {
+        logrus.Errorf("An error occurred while scanning file.\n", err)
+    }
+    return foundBreakingChanges
+}
+
+func runPreReleaseScripts(preReleaseScriptsDirpath string, releaseVersion string) error {
+	preReleaseScriptsFilepath := path.Join(preReleaseScriptsDirpath, preReleaseScriptsFilename)
+	preReleaseScriptsFile, err := os.Open(preReleaseScriptsFilepath);
+	if err != nil {
+		logrus.Errorf("An error occurred attempting to open file at provided path. Are you sure '%s' exists?", preReleaseScriptsFilepath, err)
+		return err
+	}
+	defer preReleaseScriptsFile.Close()
+	scanner := bufio.NewScanner(preReleaseScriptsFile)
+	var scripts []string
+	for scanner.Scan() {
+		scripts = append(scripts, scanner.Text())
+	}
+	for _, script := range scripts {
+		scriptCmdString := path.Join(preReleaseScriptsDirpath, script)
+		scriptCmd := exec.Command(scriptCmdString, releaseVersion)
+		err := scriptCmd.Run()
+		if err != nil {
+			logrus.Errorf("An error occurred attempting to run the following pre release script command: '%s'", scriptCmdString, err)
```

Comment:
how do we retrieve the array of command args for each pre release script file? I ask because here we're assuming every pre release script just takes in a version.

Reply:
discussed with Kevin. convo resolved